File: problem/Count_Binary_Substrings/countBinarySubstrings.js
/**
 * @param {string} s
 * @return {number}
 */
const countBinarySubstrings = s => {
    let res = 0, prev = 0, strk = 1;

    for (let i = 1; i < s.length; i++) {
        if (s[i] === s[i - 1])
            strk++;
        else {
            prev = strk;
            strk = 1;
        }

        if (strk <= prev) res++;
    }

    return res;
};

File: problem/Count_Binary_Substrings/countBinarySubstrings.c
int countBinarySubstrings(char *s)
{
    int res = 0, prev = 0, strk = 1;

    for (int i = 1; s[i] != '\0'; i++)
    {
        if (s[i] == s[i - 1])
            strk++;
        else
        {
            prev = strk;
            strk = 1;
        }
        if (strk <= prev)
            res++;
    }
    return res;
}